### ISSUE TITLE
MANIFEST file grows forever if the DB is not closed and re-opened

### ISSUE THREAD

**Issue by bgedik** [google/leveldb#299]
I have a scenario where I need to keep the DB open for a long time (say months) and process continuous updates that replace old values. So my overall data size is not changing. I have verified this: the total size of the data files is constant over time. However, the MANIFEST file size keeps growing, as compactions are periodically happening. This forces me to close/open the database periodically, since during an open, the log-structured MANIFEST file is processed and compacted. 

Shouldn't there be a upper limit for the MANIFEST file size or some period after which it is compacted, without requiring a close/open of the database.

**Comment by davidsgrogan** [comment]
Yes, that would be nice. You are running into a known limitation.

**Comment by ralphtheninja** [comment]
+1

**Comment by bgedik** [comment]
Any updates on this?

**Comment by sewe** [comment]
Same here; we also have a long running process that adds, replaces, and removes keys for month, causing the `MANIFEST` to grow and the `MANIFEST` / user-data ratio to shrink.

Worse, the `MANIFEST` may grow so large (16+ GB are not unheard of) that opening the database again fails with an out-of-memory error.

Any update on this issue would hence be very much appreciated.

**Comment by ZhangTao1596** [comment]
> Same here; we also have a long running process that adds, replaces, and removes keys for month, causing the `MANIFEST` to grow and the `MANIFEST` / user-data ratio to shrink.
> 
> Worse, the `MANIFEST` may grow so large (16+ GB are not unheard of) that opening the database again fails with an out-of-memory error.
> 
> Any update on this issue would hence be very much appreciated.

I meet this problem too. Is there any practical solutions?

**Comment by wang-yx** [comment]
I meet this problem too，Is there a solution now？

**Comment by sugumaranericsson** [comment]
i too meet this issue... any workaround ?

**Comment by ShooterIT** [comment]
Hi @cmumford Could you hav a look for this old and big trouble.
